### ISSUE TITLE
Issue with --tz-aware option

### ISSUE THREAD

**Issue by Sekhar-jami** [mongodb-labs/mongo-connector#661]
I have the following date field in mongo:

"postedDate" : ISODate("2013-12-23T00:00:00.000+0000")

But even when I run mongo-connector using the --tz-aware option, when I check the logs, it is being parsed as "2013-12-23T00:00:00+00:00"  whereas I would like to use the exact original value. How can I achieve this?

**Comment by ShaneHarvey** [comment]
Those two dates do represent the exact same time. 

Some explanation on what's actually going on under the hood here: In the mongo shell an `ISODate` is stored as a 64 bit integer representing the number of milliseconds since the Unix epoch. PyMongo represents this as a [`datetime.datetime`](https://docs.python.org/3/library/datetime.html#datetime.datetime) with or without a timezone depending on the configuration. Mongo-connector sends this `datetime.datetime` to Elasticsearch and the [elasticsearch-py](http://elasticsearch-py.readthedocs.io/en/2.4.0/) converts it to the date string in the format "2013-12-23T00:00:00+00:00" when a timezone is present or  "2013-12-23T00:00:00" when there's no timezone. You can test this out with this example:
```
$ python3 -c '
from datetime import datetime, timezone
from elasticsearch import Elasticsearch
es = Elasticsearch()

doc = {
    "date_no_timezone": datetime.now(),
    "date_utc_timezone": datetime.now(timezone.utc)
}
es.index(index="test-index", doc_type="tweet", id=1, body=doc)
res = es.get(index="test-index", doc_type="tweet", id=1)
print(res["_source"])'
{'date_no_timezone': '2017-02-21T16:48:02.071451', 'date_utc_timezone': '2017-02-22T00:48:02.071461+00:00'}
```

All this is to say that mongo-connector is doing the right thing here and those dates represent the same time.

**Comment by vaibhavbhalla07** [comment]
Agreed with the above explanation. But while conversion I wanted a format like this 
"2017-02-22T00:48:02.071Z" instead of "2017-02-22T00:48:02.071461+00:00". 

Can we have custom datetime format settings with mongo-connector?

**Comment by ShaneHarvey** [comment]
You might be able to change the date format using [Elasticsearch's ingest node preprocessing](https://www.elastic.co/guide/en/elasticsearch/reference/master/ingest.html). See: https://www.elastic.co/guide/en/elasticsearch/reference/master/date-processor.html. It might require using a Painless script processor to parse and re-format the date string: https://www.elastic.co/guide/en/elasticsearch/reference/master/script-processor.html

**Comment by ShaneHarvey** [comment]
Otherwise, we'll review adding a date format option. I don't think it will be difficult since we already format documents. It would be as simple as adding an optional conversion to the formatter: https://github.com/mongodb-labs/mongo-connector/blob/2.5.1/mongo_connector/doc_managers/formatters.py#L98
